### PR TITLE
fix(telemetry): Using a single event for analytics tracking

### PR DESCRIPTION
**Pull request by jyasveer**
- This PR uses a single event name to listen all the analytics tracking events
- Depends on https://github.com/fabric8-ui/fabric8-ui/pull/3130
- Depends on https://github.com/fabric8-launcher/ngx-launcher/pull/335. Before merging this PR the launcher package has to be bumped up
- Version upgrade for launcher to 1.0.8 to address some minor bugs

Addresses issue - https://github.com/openshiftio/openshift.io/issues/3239

### CODE REVIEW

File: src/app/shared/analytics.service.ts [hunk 1]
```diff
@@ -16,6 +16,26 @@ declare global {
...
+    'completeDependencyEditorStep': 'dependency editor completed',
+    'completePipelineStep_Create': 'pipeline completed in create',
+    'completeGitProviderStep_Create': 'git provider completed in create',
+    'completeSummaryStep_Create': 'summary completed in create',
+    'completePipelineStep_Import': 'pipeline completed in import',
+    'completeGitProviderStep_Import': 'git provider completed in import',
+    'completeSummaryStep_Import': 'summary completed in import',
+    'viewApplicationButtonClicked': 'view application button clicked',
+    'stepIndicatorClicked': 'step indicator clicked',
+    'stepIndicatorProjectInputClicked': 'step indicator project name clicked'
```
Comment (bartoszmajsak) [inline src/app/shared/analytics.service.ts]: Can't this mapping be on the `ngx-launcher` side? This way you still couple both code bases.

**Comment by joshuawilson** [comment]
please rebase to get the migration out of this PR

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @jyasveer for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

**Comment by invincibleJai** [comment]
@jyasveer fetched locally, things are looking fine.

**Comment by jyasveer** [comment]
Thanks @invincibleJai

**Comment by joshuawilson** [comment]
why no tests?

**Comment by jyasveer** [comment]
@joshuawilson There is nothing substantial that I can test. I have added only some listeners

**Comment by jyasveer** [comment]
@joshuawilson can you please merge this PR

**Comment by jyasveer** [comment]
@joshuawilson I have updated the unit tests. PTAL and merge if all is okay

**Comment by invincibleJai** [comment]
@joshuawilson pls take a look.

**Comment by jyasveer** [comment]
@joshuawilson can you please have a look and merge the PR ?

**Comment by jyasveer** [comment]
Thanks @joshuawilson

**Comment by aptmac** [comment]
Ignore my revert branch, I went to hit "hide details" and clicked "revert" by mistake.

Having said that, this PR broke unit tests.

**Comment by jiekang** [comment]
@jyasveer Can you take a look? The test runner is crashing from the tests in `analytics.service.spec.ts`

**Comment by jyasveer** [comment]
Sure let me see what is the problem

**Comment by aptmac** [comment]
The problem is here [[0]](https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/shared/analytics.service.spec.ts#L64), because of [[1]](https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/shared/analytics.service.ts#L87).

[0] https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/shared/analytics.service.spec.ts#L64
[1] https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/shared/analytics.service.ts#L87

**Pull request merged**